lnwallet/rpcwallet/rpcwallet.go

```diff
@@ -73,14 +83,104 @@ func NewRPCKeyRing(watchOnlyWallet keychain.SecretKeyRing,
 	}
 
 	return &RPCKeyRing{
-		watchOnlyWallet: watchOnlyWallet,
-		rpcTimeout:      rpcTimeout,
-		lnClient:        lnrpc.NewLightningClient(rpcConn),
-		signerClient:    signrpc.NewSignerClient(rpcConn),
-		walletClient:    walletrpc.NewWalletKitClient(rpcConn),
+		WalletController: watchOnlyWalletController,
+		watchOnlyKeyRing: watchOnlyKeyRing,
+		rpcTimeout:       rpcTimeout,
+		lnClient:         lnrpc.NewLightningClient(rpcConn),
+		signerClient:     signrpc.NewSignerClient(rpcConn),
+		walletClient:     walletrpc.NewWalletKitClient(rpcConn),
 	}, nil
 }
 
+// NewAddress returns the next external or internal address for the
+// wallet dictated by the value of the `change` parameter. If change is
+// true, then an internal address should be used, otherwise an external
+// address should be returned. The type of address returned is dictated
+// by the wallet's capabilities, and may be of type: p2sh, p2wkh,
+// p2wsh, etc. The account parameter must be non-empty as it determines
+// which account the address should be generated from.
+func (r *RPCKeyRing) NewAddress(addrType lnwallet.AddressType, change bool,
+	account string) (btcutil.Address, error) {
+
+	ctxt, cancel := context.WithTimeout(context.Background(), r.rpcTimeout)
+	defer cancel()
+
+	rpcAddrType := walletrpc.AddressType_WITNESS_PUBKEY_HASH
+	if addrType == lnwallet.NestedWitnessPubKey {
+		rpcAddrType = walletrpc.AddressType_NESTED_WITNESS_PUBKEY_HASH
+	}
+
+	remoteAddr, err := r.walletClient.NextAddr(ctxt, &walletrpc.AddrRequest{
+		Account: account,
+		Type:    rpcAddrType,
+		Change:  change,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error deriving address on remote "+
+			"signer instance: %v", err)
+	}
+
+	localAddr, err := r.WalletController.NewAddress(
```

Comment:
In the future, I think we could extra this wallet sync logic to a higher layer, but likely good enough for it to live here now.